include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,268 @@
+  //! @}
+  //====================================================================================================================
+  template<typename Tag, typename... Specs>
+  struct supports : Specs...
+  {
+    protected:
+    using Specs::process_option...;
+
+    public:
+    // Aggregate all defaults from all Specs
+    EVE_FORCEINLINE static constexpr auto defaults() noexcept
+    {
+      return  kumi::fold_left ( [](auto const& acc, auto const& e) { return options{rbr::merge(e,acc)}; }
+                              , kumi::make_tuple(Specs::defaults()...)
+                              );
+    }
+
+    //==================================================================================================================
+    //! @brief Adds an option to current callable
+    //!
+    //! If the option `o` is supported by one of the specifications `Spec`, build a new @callable that will behave as
+    //! an instance of `Tag` except the option `o` is passed inside an eve::options as an additional parameter.
+    //!
+    //! Does not participate in overload resolution if `o` is not support by any option specifications.
+    //!
+    //! @param o Option to add to current's @callable setup
+    //! @return A eve::decorated_callable behaving as `Tag` but with the options `o` set.
+    //==================================================================================================================
+    template<typename O>
+    EVE_FORCEINLINE auto operator[](O const& o) const
+    -> decltype( detail::decorated_fn<Tag,decltype(process_option(defaults(), o)),Specs...>(process_option(defaults(), o)) )
+    {
+      auto opts = process_option(defaults(), o);
+      return detail::decorated_fn<Tag,decltype(opts),Specs...>(opts);
+    }
+  };
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct conditional
+  //!   @brief Option specification for decoration via conditional value and expressions
+  //!
+  //!   eve::conditional is an option specification that can be used when defining a eve::callable type to make it
+  //!   supports decoration via `bool`, eve::logical, any eve::conditional_expr or any eve::relative_conditional_expr.
+  //!
+  //!   @groupheader{Example}
+  //!
+  //!   @godbolt{doc/traits/callable_supports.cpp}
+  //! @}
+  //====================================================================================================================
+  struct conditional
+  {
+    auto process_option(auto const& base, rbr::concepts::exactly<condition> auto opt) const
+    {
+      return options{rbr::merge(options{opt}, base)};
+    }
+
+    template<std::same_as<bool> O>
+    auto process_option(auto const& base, O opt) const
+    {
+      // Just delay the evaluation of the type by injecting some templates
+      using type = std::conditional_t<std::same_as<bool,O>,std::uint8_t,O>;
```

Comment:
Confused as to why

Reply:
If I put logical<std::uint8_t> here directly, it instanciates stuff too early.

Reply:
Ah - like logical<std::uint8_t>? ok